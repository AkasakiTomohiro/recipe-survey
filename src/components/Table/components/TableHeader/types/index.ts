/**
 * @file index.ts
 * @author Artan's Projects
 * @copyright © 2024 Artan's Projects. All rights reserved.
 */

export interface ITableHeaderProps {

  /**
   * ヘッダーの文字列
   */
  headers: string[];

  /**
   * Collapsible表示があるか
   */
  isCollapse?: boolean;
}
