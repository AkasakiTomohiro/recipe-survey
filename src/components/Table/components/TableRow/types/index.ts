/**
 * @file index.ts
 * @author Artan's Projects
 * @copyright © 2024 Artan's Projects. All rights reserved.
 */

export interface ITableRowProps {

  /**
   * 行番号
   */
  index: number;

  /**
   * 1行のデータ
   */
  row: (number | string)[];
}
