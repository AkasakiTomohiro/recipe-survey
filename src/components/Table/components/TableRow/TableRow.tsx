/**
 * @file TableRow.tsx
 * @author Artan's Projects
 * @copyright Â© 2024 Artan's Projects. All rights reserved.
 */

import TableCell from '@mui/material/TableCell';
import MuiTableRow from '@mui/material/TableRow';

import { ITableRowProps } from './types';

export const TableRow = ({ row, index }: ITableRowProps): JSX.Element => {
  return (
    <MuiTableRow hover>
      {row.map((m, i) => (<TableCell key={`table-row-${index}-cell-${i}`}>{m}</TableCell>))}
    </MuiTableRow>
  );
};
