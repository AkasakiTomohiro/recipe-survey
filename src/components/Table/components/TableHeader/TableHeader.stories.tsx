/**
 * @file TableHeader.stories.tsx
 * @author Artan's Projects
 * @copyright Â© 2024 Artan's Projects. All rights reserved.
 */

import type { Meta, StoryObj } from '@storybook/react';

import Paper from '@mui/material/Paper';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableRow from '@mui/material/TableRow';

import { TableHeader } from '../TableHeader/TableHeader';

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
const meta = {
  title     : 'components/TableHeader',
  component : TableHeader,
  parameters: {

    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
    layout: 'centered'
  },

  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs']
} satisfies Meta<typeof TableHeader>;

export default meta;
type Story = StoryObj<typeof meta>;

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args
export const Primary: Story = {
  args: {
    headers: ['Header1', 'Header2', 'Header3'],
    isCollapse: false
  },
  decorators: [Story => (
    <TableContainer component={Paper}>
      <Story />
      <TableRow>
        <TableCell>Value1</TableCell>
        <TableCell>Value2</TableCell>
        <TableCell>Value3</TableCell>
      </TableRow>
    </TableContainer>
  )]
};

export const CollapseHeader: Story = {
  args: {
    headers: ['Header1', 'Header2', 'Header3'],
    isCollapse: true
  },
  decorators: [Story => (
    <TableContainer component={Paper}>
      <Story />
      <TableRow>
        <TableCell></TableCell>
        <TableCell>Value1</TableCell>
        <TableCell>Value2</TableCell>
        <TableCell>Value3</TableCell>
      </TableRow>
    </TableContainer>
  )]
};
