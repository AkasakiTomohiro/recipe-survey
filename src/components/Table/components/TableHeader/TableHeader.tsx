/**
 * @file TableHeader.tsx
 * @author Artan's Projects
 * @copyright Â© 2024 Artan's Projects. All rights reserved.
 */

import TableCell from '@mui/material/TableCell';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';

import { ITableHeaderProps } from './types';

export const TableHeader = (props: ITableHeaderProps): JSX.Element => {
  return (
    <TableHead>
      <TableRow>
        {props.isCollapse && <TableCell />}
        {props.headers.map((m, i) => (<TableCell align="left" key={`table-header-${i}`}>{m}</TableCell>))}
      </TableRow>
    </TableHead>
  );
};
