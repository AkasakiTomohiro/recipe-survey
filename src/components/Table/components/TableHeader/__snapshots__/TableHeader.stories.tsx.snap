// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Table/TableHeader CollapseHeader smoke-test 1`] = `
<div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-xpjyhy-MuiPaper-root-MuiTableContainer-root">
  <thead class="MuiTableHead-root css-15wwp11-MuiTableHead-root">
    <tr class="MuiTableRow-root MuiTableRow-head css-dvunen-MuiTableRow-root">
      <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-epgjbe-MuiTableCell-root"
          scope="col"
      >
      </th>
      <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-epgjbe-MuiTableCell-root"
          scope="col"
      >
        Header1
      </th>
      <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-epgjbe-MuiTableCell-root"
          scope="col"
      >
        Header2
      </th>
      <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-epgjbe-MuiTableCell-root"
          scope="col"
      >
        Header3
      </th>
    </tr>
  </thead>
  <tr class="MuiTableRow-root css-dvunen-MuiTableRow-root">
    <td class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root">
    </td>
    <td class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root">
      Value1
    </td>
    <td class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root">
      Value2
    </td>
    <td class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root">
      Value3
    </td>
  </tr>
</div>
`;

exports[`components/Table/TableHeader Primary smoke-test 1`] = `
<div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-xpjyhy-MuiPaper-root-MuiTableContainer-root">
  <thead class="MuiTableHead-root css-15wwp11-MuiTableHead-root">
    <tr class="MuiTableRow-root MuiTableRow-head css-dvunen-MuiTableRow-root">
      <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-epgjbe-MuiTableCell-root"
          scope="col"
      >
        Header1
      </th>
      <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-epgjbe-MuiTableCell-root"
          scope="col"
      >
        Header2
      </th>
      <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-epgjbe-MuiTableCell-root"
          scope="col"
      >
        Header3
      </th>
    </tr>
  </thead>
  <tr class="MuiTableRow-root css-dvunen-MuiTableRow-root">
    <td class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root">
      Value1
    </td>
    <td class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root">
      Value2
    </td>
    <td class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root">
      Value3
    </td>
  </tr>
</div>
`;
